In [158]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None) 

In [159]:
Pop = pd.read_csv("Population.csv", delimiter=",")
ManageDrinking = pd.read_csv("BasicAndSafelyManagedDrinkingWaterServices.csv", delimiter=",")
Mortality = pd.read_csv("MortalityRate copie.csv", delimiter=";")
Politic =  pd.read_csv("PoliticalStability.csv", delimiter=",")
Region = pd.read_csv("RegionCountry.csv", delimiter=",")
#UrbanRate = pd.read_csv("Complete2.csv", delimiter=",")
#display(Region[:10], Region.shape)

### Nettoyage fichier Country Regions

In [160]:
Region.rename(columns={"COUNTRY (DISPLAY)": "Country", "REGION (DISPLAY)": "Region"}, inplace=True)
print("Shape fichier region",Region.shape)
print("Valeurs uniques du fichier region",'\n',Region.nunique())

Shape fichier region (194, 2)
Valeurs uniques du fichier region 
 Region       6
Country    194
dtype: int64


Le fichier Region possede 194 lignes pour 194 Country et 6 region. Chaque pays n'apparait donc qu'une seule fois

### Fichier Population

In [161]:
# Supression des lignes "Male" et "Female"

Pop = Pop.loc[Pop["Granularity"]!= "Male"]
Pop = Pop.loc[Pop["Granularity"]!= "Female"]

print("Shape de pop:",Pop.shape)
print(Pop.nunique())


Shape de pop: (13258, 4)
Country          239
Granularity        3
Year              19
Population     12900
dtype: int64


### Fusion Fichier population et regions

In [162]:
Complete_Df = pd.merge(Pop, Region, on="Country", how="right")
display(Complete_Df[:1])
print("Shape de complete",Complete_Df.shape,'\n')
print("Valeurs uniques de Complete_df",'\n',Complete_Df.nunique())

Country Granularity    Year  Population  Region
0  Albania       Total  2000.0    3129.243  Europe

Shape de complete (10894, 5) 

Valeurs uniques de Complete_df 
 Country          194
Granularity        3
Year              19
Population     10803
Region             6
dtype: int64


### Fichier Politic stability

In [163]:

Politic.info()
Politic.nunique()



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3526 entries, 0 to 3525
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Country              3526 non-null   object 
 1   Year                 3526 non-null   int64  
 2   Political_Stability  3526 non-null   float64
 3   Granularity          3526 non-null   object 
dtypes: float64(1), int64(1), object(2)
memory usage: 110.3+ KB


Country                200
Year                    18
Political_Stability    444
Granularity              1
dtype: int64

le fichhier de stabilité politique traite 18 années (entre 2000 et 2018). Atention il manque l'année 2001.
Il y a 200 pays donc 6 de plus que dans le fichier population

### Fusion de Complete_DF Political Stability

In [164]:
Complete_Df= pd.merge(Complete_Df, Politic, on=["Country", "Year", "Granularity"], how="left")

#Complete_Df.rename(columns={"Political_Stability_x": "Political_Stability"}, inplace=True)
#Complete_Df.drop(labels="Political_Stability_y", axis=1,inplace=True)
display(Complete_Df[:5])
print("Shape de Complet_df",Complete_Df.shape)
print(Complete_Df.nunique())

Country Granularity    Year  Population  Region  Political_Stability
0  Albania       Total  2000.0    3129.243  Europe                -0.54
1  Albania       Rural  2000.0    1818.833  Europe                  NaN
2  Albania       Urban  2000.0    1303.137  Europe                  NaN
3  Albania       Total  2001.0    3129.704  Europe                  NaN
4  Albania       Rural  2001.0    1797.423  Europe                  NaN

Shape de Complet_df (10894, 6)
Country                  194
Granularity                3
Year                      19
Population             10803
Region                     6
Political_Stability      437
dtype: int64


Apres le Merge les pays en plus qui etait present dans le fichier de stabilite politique ont été supprimé

### Fichier Mortality Rate

In [165]:
print("Shape Mortality Rate",Mortality.shape)
print("\nValeur unique dans Mortality rate",'\n',Mortality.nunique())
display(Mortality[:1])
print(Complete_Df.nunique())

Shape Mortality Rate (183, 5)

Valeur unique dans Mortality rate 
 Year                                                               1
Country                                                          183
Granularity                                                        1
Mortality rate attributed to exposure to unsafe WASH services    183
WASH d                                                           183
dtype: int64


Year      Country Granularity  \
0  2016  Afghanistan       Total   

   Mortality rate attributed to exposure to unsafe WASH services    WASH d  
0                                           13.92067              4824.353

Country                  194
Granularity                3
Year                      19
Population             10803
Region                     6
Political_Stability      437
dtype: int64


Le fichier de taux de mortalité couvre 183 pays pour l'année 2016. 

### Fusion entre Complete_Df et le fichier "Mortality rate"

In [166]:
Complete_Df= pd.merge(Complete_Df, Mortality, on=["Country", "Year", "Granularity"], how="left")
#Complete_Df.drop(labels="Political_Stability_y", axis=1,inplace=True)
#display(Complete_Df.loc[Complete_Df["Year"]== 2016])
print(Complete_Df.shape)
display(Complete_Df[:1])
print(Complete_Df.nunique())
print(Complete_Df.isna())



(10894, 8)


Country Granularity    Year  Population  Region  Political_Stability  \
0  Albania       Total  2000.0    3129.243  Europe                -0.54   

   Mortality rate attributed to exposure to unsafe WASH services  WASH d  
0                                                NaN                 NaN

Country                                                            194
Granularity                                                          3
Year                                                                19
Population                                                       10803
Region                                                               6
Political_Stability                                                437
Mortality rate attributed to exposure to unsafe WASH services      176
WASH d                                                             176
dtype: int64
       Country  Granularity   Year  Population  Region  Political_Stability  \
0        False        False  False       False   False                False   
1        False        False  False       False   False                 True   
2        False        False  False       False   False                 True   
3        False        False  False       False   False                 True   
4        False        Fa

### Fichier Basic and safely managed watter drinking services

In [167]:
print("Shape ManageDrinking",ManageDrinking.shape)
print("\nValeur unique dans ManegeDrinking",'\n',ManageDrinking.nunique())
display(ManageDrinking[:3])

Shape ManageDrinking (10476, 5)

Valeur unique dans ManegeDrinking 
 Year                                                             18
Country                                                         194
Granularity                                                       3
Population using at least basic drinking-water services (%)    7704
Population using safely managed drinking-water services (%)    2877
dtype: int64


Year      Country Granularity  \
0  2000  Afghanistan       Rural   
1  2000  Afghanistan       Total   
2  2000  Afghanistan       Urban   

   Population using at least basic drinking-water services (%)  \
0                                           21.61913             
1                                           27.77190             
2                                           49.48745             

   Population using safely managed drinking-water services (%)  
0                                                NaN            
1                                                NaN            
2                                                NaN

### Fusion entre Complete et Managed water drinking services

In [168]:
print(Complete_Df.nunique())
Complete_Df= pd.merge(Complete_Df, ManageDrinking, on=["Country", "Year", "Granularity"], how="left")
#Complete_Df.drop(labels="Political_Stability_y", axis=1,inplace=True)

print(Complete_Df.shape)
display(Complete_Df[:3])
print(Complete_Df.nunique())

Country                                                            194
Granularity                                                          3
Year                                                                19
Population                                                       10803
Region                                                               6
Political_Stability                                                437
Mortality rate attributed to exposure to unsafe WASH services      176
WASH d                                                             176
dtype: int64
(10894, 10)


Country Granularity    Year  Population  Region  Political_Stability  \
0  Albania       Total  2000.0    3129.243  Europe                -0.54   
1  Albania       Rural  2000.0    1818.833  Europe                  NaN   
2  Albania       Urban  2000.0    1303.137  Europe                  NaN   

   Mortality rate attributed to exposure to unsafe WASH services  WASH d  \
0                                                NaN                 NaN   
1                                                NaN                 NaN   
2                                                NaN                 NaN   

   Population using at least basic drinking-water services (%)  \
0                                           87.86662             
1                                           81.78472             
2                                           96.35529             

   Population using safely managed drinking-water services (%)  
0                                           49.29324            
1                                                NaN            
2                                                NaN

Country                                                            194
Granularity                                                          3
Year                                                                19
Population                                                       10803
Region                                                               6
Political_Stability                                                437
Mortality rate attributed to exposure to unsafe WASH services      176
WASH d                                                             176
Population using at least basic drinking-water services (%)       7617
Population using safely managed drinking-water services (%)       2817
dtype: int64


### Calcul du taux d'habitant en zone urbaine par pays et par années

In [169]:
# Recuperation du nombre d'habitant zone urbaine par pays et par années
urban_rate= Complete_Df.loc[ Complete_Df["Granularity"] == "Urban"] 

# Recuperation du nombre d'habitant total par pays et par années
Pop.drop_duplicates(subset=["Country","Year"], inplace=True)
display(Pop[:5])

#Creation d'un DataFrame ou reunir toutes les valeurs
urban_rate2 = pd.merge(urban_rate, Pop, on=["Country","Year"], how="left")

#Calcul du taux d'habitant en zone urbaine 
urban_rate2["Rate pop urban"] = round((urban_rate2["Population_x"] * 100) / urban_rate2["Population_y"], 1)


display(urban_rate2[:1])
urban_rate2.shape

Country Granularity  Year  Population
0   Afghanistan       Total  2000   20779.953
5   Afghanistan       Total  2001   21606.988
10  Afghanistan       Total  2002   22600.770
15  Afghanistan       Total  2003   23680.871
20  Afghanistan       Total  2004   24726.684

Country Granularity_x    Year  Population_x  Region  Political_Stability  \
0  Albania         Urban  2000.0      1303.137  Europe                  NaN   

   Mortality rate attributed to exposure to unsafe WASH services  WASH d  \
0                                                NaN                 NaN   

   Population using at least basic drinking-water services (%)  \
0                                           96.35529             

   Population using safely managed drinking-water services (%) Granularity_y  \
0                                                NaN                   Total   

   Population_y  Rate pop urban  
0      3129.243            41.6

(3631, 13)

In [170]:
# Nettoyage de urban rate pour n'avoir plus que les colonne a fusionner avec complete_df

urban_rate2.drop(['Political_Stability', 'Region',"Mortality rate attributed to exposure to unsafe WASH services",
"WASH d","Population using at least basic drinking-water services (%)","Granularity_y",
"Population using safely managed drinking-water services (%)","Population_x","Population_y"], axis=1, inplace=True)
urban_rate2.rename(columns={"Granularity_x":"Granularity"},inplace=True)
display(urban_rate2[:5])
print(Complete_Df.nunique())

Country Granularity    Year  Rate pop urban
0  Albania       Urban  2000.0            41.6
1  Albania       Urban  2001.0            42.3
2  Albania       Urban  2002.0            43.4
3  Albania       Urban  2003.0            44.5
4  Albania       Urban  2004.0            45.5

Country                                                            194
Granularity                                                          3
Year                                                                19
Population                                                       10803
Region                                                               6
Political_Stability                                                437
Mortality rate attributed to exposure to unsafe WASH services      176
WASH d                                                             176
Population using at least basic drinking-water services (%)       7617
Population using safely managed drinking-water services (%)       2817
dtype: int64


### ajout de la colonne du taux de la population vivant en zone urbaine("rate pop urban") a completeDf

In [171]:
Complete_Df= pd.merge(Complete_Df, urban_rate2, on=["Country", "Year", "Granularity"], how="left")
display(Complete_Df[:10])
print(Complete_Df.shape)

Country Granularity    Year  Population  Region  Political_Stability  \
0  Albania       Total  2000.0    3129.243  Europe                -0.54   
1  Albania       Rural  2000.0    1818.833  Europe                  NaN   
2  Albania       Urban  2000.0    1303.137  Europe                  NaN   
3  Albania       Total  2001.0    3129.704  Europe                  NaN   
4  Albania       Rural  2001.0    1797.423  Europe                  NaN   
5  Albania       Urban  2001.0    1324.985  Europe                  NaN   
6  Albania       Total  2002.0    3126.187  Europe                -0.29   
7  Albania       Rural  2002.0    1762.229  Europe                  NaN   
8  Albania       Urban  2002.0    1356.800  Europe                  NaN   
9  Albania       Total  2003.0    3118.023  Europe                -0.31   

   Mortality rate attributed to exposure to unsafe WASH services  WASH d  \
0                                                NaN                 NaN   
1                                                NaN                 NaN   
2                                                NaN                 NaN   
3                                                NaN                 NaN   
4                                                NaN                 NaN   
5                                                NaN                 NaN   
6                                                NaN                 NaN   
7                                                NaN                 NaN   
8                                                NaN                 NaN   
9                                                NaN                 NaN   

   Population using at least basic drinking-water services (%)  \
0                                           87.86662             
1                                           81.78472             
2                                           96.35529             
3                                           87.88955             
4                                           81.83538             
5                                           96.10231             
6                                           87.91309             
7                                           81.88605             
8                                           95.74100             
9                                           87.92893             

   Population using safely managed drinking-water services (%)  Rate pop urban  
0                                           49.29324                       NaN  
1                                                NaN                       NaN  
2                                                NaN                      41.6  
3                                           49.21738                       NaN  
4                                                NaN                       NaN  
5                                                NaN                      42.3  
6                                           49.08154                       NaN  
7                                                NaN                       NaN  
8                                                NaN                      43.4  
9                                           48.94225                       NaN

(10894, 11)


In [172]:
print(Complete_Df.nunique())
print("\n\n",Complete_Df.info())

Country                                                            194
Granularity                                                          3
Year                                                                19
Population                                                       10803
Region                                                               6
Political_Stability                                                437
Mortality rate attributed to exposure to unsafe WASH services      176
WASH d                                                             176
Population using at least basic drinking-water services (%)       7617
Population using safely managed drinking-water services (%)       2817
Rate pop urban                                                     883
dtype: int64
<class 'pandas.core.frame.DataFrame'>
Int64Index: 10894 entries, 0 to 10893
Data columns (total 11 columns):
 #   Column                                                         Non-Null Count  Dtype  
---  

In [173]:
Complete_Df.to_csv(r"complete_Df.csv",sep=';', index=False)
